jquery/attributes.html is flaky fail with StreamClientConnection  ASSERTION FAILED: !m_connection->isValid()

kkinnunen-apple opened the pull request:
#### c0a9a955b480bb69772ceccfa2eca68906532e7f
<pre>
jquery/attributes.html is flaky fail with StreamClientConnection  ASSERTION FAILED: !m_connection-&gt;isValid()
<a href="https://bugs.webkit.org/show_bug.cgi?id=261555">https://bugs.webkit.org/show_bug.cgi?id=261555</a>
rdar://115494068

Reviewed by Antti Koivisto.

Fix two tests that create StreamClientConnection instances through
the JS IPC Testing API. The connections must be invalidated
if they are created.

Both allow and deny tests are run as normal JS business as usual, even
though the test runner automatically ends the deny test when GPUP
crashes. This means that the JS GC will clean up the stream connections
during the next tests, causing unrelated tests to fail with the
assertion.

Fix the allow test to actually test what it tests, e.g. to fail if
WebGPU is disabled.

* LayoutTests/ipc/restrictedendpoints/allow-access-webGPU-expected.txt:
* LayoutTests/ipc/restrictedendpoints/allow-access-webGPU.html:
* LayoutTests/ipc/restrictedendpoints/deny-access-webGPU.html:
* LayoutTests/resources/ipc.js:
(randomID):
(asyncFlush):
(syncFlush): Deleted.
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::asyncPing):
(WebKit::IPCTester::syncPing):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/Shared/IPCTester.messages.in:

Canonical link: <a href="https://commits.webkit.org/268086@main">https://commits.webkit.org/268086@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2020a809d7a72ace1922df6eacf1976c14f323c9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18640 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18980 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19582 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20502 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17447 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22290 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19121 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19299 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18866 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19021 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16226 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21378 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16243 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16988 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23433 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17268 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17159 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21331 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17762 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15065 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16815 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/16820 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4430 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21182 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17601 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/platform/mac-wk1/TestExpectations:
```diff
@@ -2530,7 +2530,7 @@ imported/w3c/web-platform-tests/geolocation-API/permission.https.html [ WontFix
 
 # IPC is for WK2
-ipc/shared-video-frame-size.html [ Skip ]
+ipc [ Skip ]
```
It is slightly better to mark it as WontFix, because in theory, skipped tests are tracked as something that needs to be fixed. In practice, there is currently no difference, as we don't actively try to unskip.

kkinnunen-apple committed:
jquery/attributes.html is flaky fail with StreamClientConnection  ASSERTION FAILED: !m_connection->isValid()
https://bugs.webkit.org/show_bug.cgi?id=261555
rdar://115494068

Reviewed by Antti Koivisto.

Fix two tests that create StreamClientConnection instances through
the JS IPC Testing API. The connections must be invalidated
if they are created.

Both allow and deny tests are run as normal JS business as usual, even
though the test runner automatically ends the deny test when GPUP
crashes. This means that the JS GC will clean up the stream connections
during the next tests, causing unrelated tests to fail with the
assertion.

Fix the allow test to actually test what it tests, e.g. to fail if
WebGPU is disabled.

* LayoutTests/ipc/restrictedendpoints/allow-access-webGPU-expected.txt:
* LayoutTests/ipc/restrictedendpoints/allow-access-webGPU.html:
* LayoutTests/ipc/restrictedendpoints/deny-access-webGPU.html:
* LayoutTests/resources/ipc.js:
(randomID):
(asyncFlush):
(syncFlush): Deleted.
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::asyncPing):
(WebKit::IPCTester::syncPing):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/Shared/IPCTester.messages.in:

Canonical link: https://commits.webkit.org/268086@main
